src/polyhedralGravity/calculation/SanityCheck.cpp

```diff
@@ -0,0 +1,76 @@
+#include "SanityCheck.h"
+
+namespace polyhedralGravity::SanityCheck {
+
+    bool checkNormalsOutwardPointing(const Polyhedron &polyhedron) {
+        using namespace util;
+        auto it = GravityModel::transformPolyhedron(polyhedron);
+        // All normals have to point outwards (intersect the polyhedron even times)
+        return thrust::transform_reduce(
+                thrust::device,
+                it.first, it.second, [&polyhedron](const Array3Triplet &face) {
+                    // The centroid of the triangular face
+                    const Array3 centroid = (face[0] + face[1] + face[2]) / 3.0;
+
+                    // The normal of the plane calculated with two segments of the triangle
+                    const Array3 segmentVector1 = face[1] - face[0];
+                    const Array3 segmentVector2 = face[2] - face[1];
+                    const Array3 normal = util::normal(segmentVector1, segmentVector2);
+
+                    // The origin of the array has a slight offset in direction of the normal
+                    const Array3 rayOrigin = centroid + (normal * EPSILON);
+
+                    // If the ray intersects the polyhedron even-times than the normal points outwards
```

Comment:
```suggestion
                    // If the ray intersects the polyhedron an even number of times then the normal points outwards
```
just for clarity, had to read thrice :)